components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -937,6 +937,29 @@ public static AttributeSchema buildCustomSchema(int tenantId) throws CharonExcep
         }
     }
 
+    /**
+     * Returns SCIM2 system AttributeSchema of the tenant.
+     *
+     * @param tenantId  Tenant ID.
+     * @return scim2 system schema.
+     * @throws CharonException If an error occurred in retrieving custom schema.
+     */
+    public static AttributeSchema buildSystemSchema(int tenantId) throws CharonException {
+
+        try {
+            SCIMCustomSchemaProcessor scimCustomSchemaProcessor = new SCIMCustomSchemaProcessor();
+            List<SCIMCustomAttribute> attributes =
+                    scimCustomSchemaProcessor.getCustomAttributes(IdentityTenantUtil.getTenantDomain(tenantId),
+                            SCIMConstants.SYSTEM_USER_SCHEMA_URI);
+            AttributeSchema attributeSchema = SCIMSystemSchemaExtensionBuilder.getInstance()
+                    .buildSystemSchemaExtension(attributes);
+            SCIMCustomAttributeSchemaCache.getInstance().addSCIMCustomAttributeSchema(tenantId, attributeSchema);
```

Comment:
Is adding to custom attribute schema intended? - addSCIMCustomAttributeSchema

Reply:
Wouldn't these system schemas also be returned when retrieving custom schemas, given that we're adding system schemas to the custom attribute schema cache?

Reply:
Please verify this by retrieving a user first and then calling the SCIM2/Schemas endpoint. Let’s check what is being returned. If it correctly handles by resolving the dialect URI this is fine.

Reply:
Added cache in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/615